app/src/pages/playground/ChatTemplateDndSensors.ts

```diff
@@ -0,0 +1,85 @@
+import { Draggable, KeyboardSensor, PointerSensor } from "@dnd-kit/dom";
+
+/**
+ * Copy of @dnd-kit/dom
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L18
+ */
+type KeyCode = KeyboardEvent["code"];
+/**
+ * Copy of @dnd-kit/dom KeyboardCodes type.
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L20
+ */
+type KeyboardCodes = {
+  start: KeyCode[];
+  cancel: KeyCode[];
+  end: KeyCode[];
+  up: KeyCode[];
+  down: KeyCode[];
+  left: KeyCode[];
+  right: KeyCode[];
+};
+
+/**
+ * Copy of @dnd-kit/dom KeyboardSensorOptions type.
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L30
+ */
+interface KeyboardSensorOptions {
+  keyboardCodes?: KeyboardCodes;
+}
+
+/**
+ * A dnd-kit {@link KeyboardSensor} that ignores keyboard events when the user is typing in a text input.
+ */
+export class ChatTemplateKeyboardSensor extends KeyboardSensor {
+  protected handleStart(
+    event: KeyboardEvent,
+    source: Draggable,
+    options: KeyboardSensorOptions | undefined
+  ): void {
+    /**
+     * In some instances a text input may be nested within a draggable element.
+     * In that case we want to ignore the keyboard event to allow the user to type.
+     * In all other cases, we want to allow the user to move the draggable element.
+     */
+    if (
+      event.target instanceof HTMLInputElement ||
+      event.target instanceof HTMLTextAreaElement
+    ) {
+      return;
+    }
+    super.handleStart(event, source, options);
+  }
+}
+
+/**
+ * The default {@link PointerSensor} with activation constraints that allow for a delay and distance.
+ *
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/react/src/core/context/DragDropProvider.tsx#L99
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/manager/manager.ts#L23
+ */
+const ChatTemplatePointerSensor = PointerSensor.configure({
```

Comment:
without doing this, clicking anywhere on the li caused the whole thing to flicker and started a movement command. THis is the default pulled from the links above, I tried different ways of getting all the defaults (except the keyboard) but it wasn't possible so this seemed like the best approach